exporter/SynthesisFusionAddin/proto/deps.py

```diff
@@ -1,18 +1,80 @@
 import platform, subprocess
 from pathlib import Path
-from os import __file__, path
+# from os import __file__, path, system
+import os
+from src.general_imports import INTERNAL_ID;
+
+help("modules")
 
 import adsk.core, adsk.fusion
 
+system = platform.system()
+
+def getPythonFolder() -> str:
+    """Retreives the folder that contains the Autodesk python executable
+
+    Raises:
+        ImportError: Unrecognized Platform
+
+    Returns:
+        str: The path that contains the Autodesk python executable
+    """
+
+    # Thank you Kris Kaplan
+    import sys
+    import importlib.machinery
+    osPath = importlib.machinery.PathFinder.find_spec('os', sys.path).origin
+
+    if system == "Windows":
+        pythonFolder = Path(osPath).parents[
+            1
+        ]  # Assumes the location of the fusion python executable is two folders up from the os lib location
+    elif system == "Darwin":
+        pythonFolder = f'{Path(osPath).parents[2]}/bin'
+    else:
+        raise ImportError(
+            f"Unsupported platform! This add-in only supports windows and macos"
+        )
+    
+    logging.getLogger(f'{INTERNAL_ID}').debug(f'Python Folder -> {pythonFolder}')
+    return pythonFolder
+
+def executeCommand(command: tuple) -> int:
+    """Abstracts the execution of commands to account for platform differences
+
+    Args:
+        command (tuple): Tuple starting with command, and each indice having the arguments for said command
+
+    Returns:
+        int: Exit code of the process
+    """
+    if system == "Windows":
+        executionResult = subprocess.call(
+            command,
+            bufsize=1,
+            creationflags=subprocess.CREATE_NO_WINDOW,
+            shell=False
+        )
+    else:
+        # executionResult = subprocess.call(
+        #     command,
+        #     bufsize=1,
+        #     shell=False
+        # )
+        joinedCommand = str.join(' ', command)
+        executionResult = os.system(joinedCommand)
```

Comment:
Are we planning on ignoring the security workflow?

Reply:
Yeah, there's no good way around it. The complaint is that there is possible command injection into the shell that python will use to run our pip commands. However, it doesn't take into account that we don't give the user an opportunity to inject anything, and if they really really wanted to, you can modify it because it's Python. At the end of the day we control the release and as long as we don't allow users to specify custom dependencies, then we're fine.